src/prototype_creep_resources.js

```diff
@@ -470,14 +470,21 @@ Creep.prototype.getEnergy = function() {
     return false;
   }
 
-  if (this.getDroppedEnergy()) {
-    return true;
+  if (this.room.storage) {
+    if (this.getDroppedEnergy(res => res.amount > this.carryCapacity ||
+        res.amount > this.room.storage.store.energy || this.pos.getRangeTo(res) * 2 < this.pos.getRangeTo(this.room.storage))) {
+      return true;
+    }
```

Comment:
Hm, I don't like this so much.

I think we should get the droppedEnergy first, if it makes sense. Especially because droppedEnergy decays, energy in the storage not. If not get it from the storage.
Instead of working with conditions here, I would prefer if we could tweak the formulas in 
`src/prototype_creep_startup_tasks.js` line 384.

So the idea is to have some kind of calculation if it makes sense to move to that droppedEnergy.